bot/persistence/database_connector.py

```diff
@@ -101,9 +99,7 @@ def get_member_warnings(self, user_id: int) -> Optional[List[tuple]]:
             result = db_manager.execute(queries.GET_MEMBER_WARNINGS, (user_id,))
 
             rows = result.fetchall()
-            if rows:
-                return rows
-            return None
+            return rows
```

Comment:
fetchall() returns an empty list if no elements are found. This is not the same as None. Therefore on the one hand you change behavior, which might break existing code that calls this function (did you check that?) and on the other hand, the mypy return type is wrong, as it should no longer be `Optional[List[tuple]]` but List[tuple]`

Reply:
For Python, both an empty list and None evaluate to `false` if they are being used in an if statement. I've checked checked the code and everything should work as expected.

You're right with the return type. Thanks!

Reply:
Just saying bc  None == [] is false. So if the code somewhere had sth like  len(rows)==0 then it would break.